### PR TITLE
add tri state verification to slack (not slack webhook)

### PR DESCRIPTION
**Pull request by zubairk14**
### Description:
Explain the purpose of the PR.

### Checklist:
* [x] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/slack/slack.go [hunk 1]
```diff
@@ -53,34 +56,54 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
...
 					defer res.Body.Close()
 					var authResponse authRes
 					if err := json.NewDecoder(res.Body).Decode(&authResponse); err != nil {
 						continue
 					}
-					s.Verified = authResponse.Ok
+
+					if authResponse.Ok {
+						s1.Verified = true
+						// Slack API returns 200 even if the token is invalid. We need to check the error field.
```
Comment (rosecodym) [inline pkg/detectors/slack/slack.go:87]: do you think it's worth adding some code to check for non-2xx responses?

**Review by rosecodym — CHANGES_REQUESTED** [review]
you're a machine! (only change request is to remove the commented stuff)

File: pkg/detectors/slack/slack.go (hunk 1 above)
Comment (zubairk14) [reply pkg/detectors/slack/slack.go:87]: this API seems to return 200 always so imo our best bet is to depend on the else block for whenever the `authResponse` is not OK or the else to catch when the client response catches an error

```
 else {
						s1.VerificationError = fmt.Errorf("unexpected error auth response %+v", authResponse.Error)
					}
					```

Comment (rosecodym) [reply pkg/detectors/slack/slack.go:87]: What about checking for a response body that can't be successfully decoded as JSON?

File: pkg/detectors/slack/slack.go [hunk 2]
```diff
@@ -53,34 +56,54 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
...
+					} else if authResponse.Error == "invalid_auth" {
+						// The secret is determinately not verified (nothing to do)
+					} else {
+						s1.VerificationError = fmt.Errorf("unexpected error auth response %+v", authResponse.Error)
```
Comment (rosecodym) [inline pkg/detectors/slack/slack.go:91]: Can you add a test for this? One of the fake HTTP clients available allows you to inject a fake response body - lmk if you want me to dig up a specific example.

File: pkg/detectors/slack/slack.go (hunk 1 above)
Comment (dustin-decker) [reply pkg/detectors/slack/slack.go:87]: Added

**Pull request merged**